src/frontend/settingsdialogs/bttexttospeechsettings.cpp

```diff
@@ -0,0 +1,152 @@
+/*********
+*
+* In the name of the Father, and of the Son, and of the Holy Spirit.
+*
+* This file is part of BibleTime's source code, https://bibletime.info/
+*
+* Copyright 2024 by the BibleTime developers.
+* The BibleTime source code is licensed under the GNU General Public License
+* version 2.0.
+*
+**********/
+
+#ifdef BUILD_TEXT_TO_SPEECH
+
+#include "bttexttospeechsettings.h"
+
+#include <QComboBox>
+#include <QFormLayout>
+#include <QLabel>
+#include <QLocale>
+#include <QVoice>
+#include "../../backend/config/btconfig.h"
+#include "../../util/btconnect.h"
+#include "../../util/cresmgr.h"
+#include "../bibletime.h"
+#include "cconfigurationdialog.h"
+
+
+BtTextToSpeechSettingsPage::BtTextToSpeechSettingsPage(CConfigurationDialog *parent)
+        : BtConfigDialog::Page(CResMgr::settings::audio::icon(), parent) {
+    setHeaderText(tr("Text-to-speech"));
+
+    QVBoxLayout* mainLayout = new QVBoxLayout(this);
+    QFormLayout* formLayout = new QFormLayout();
+
+    const QStringList availableEngines = QTextToSpeech::availableEngines();
+    if (availableEngines.isEmpty()) {
+        formLayout->addWidget(new QLabel(tr("No Qt text-to-speech plug-ins were found. Please install a plugin and restart the application."), this));
```

Comment:
Please try to adhere to a 80-character line length limit.

Reply:
Done